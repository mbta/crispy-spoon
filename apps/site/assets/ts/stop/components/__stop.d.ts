import {
  Direction,
  DirectionId,
  EnhancedRoute,
  Stop,
  Alert
} from "../../__v3api";
import { MapData } from "../../leaflet/components/__mapdata";

export interface TypedRoutes {
  group_name: string;
  routes: RouteWithDirections[];
}

export interface StopMapData {
  map_data: MapData;
  map_srcset: string;
  map_url: string;
}

export interface RouteWithDirection {
  direction_id: DirectionId | null;
  route: EnhancedRoute;
}

export interface SuggestedTransfer {
  stop: Stop;
  distance: number;
  routes_with_direction: RouteWithDirection[];
}

export interface StopPageData {
  stop: Stop;
  street_view_url: string | null;
  routes: TypedRoutes[];
  routes_with_direction?: RouteWithDirection[];
  tabs: Tab[];
  zone_number: string;
  alerts: Alert[];
  suggested_transfers: SuggestedTransfer[];
  retail_locations: RetailLocationWithDistance[];
  tab: string;
  alerts_tab: AlertsTab;
  routes_and_alerts: { [key: string]: Alert[] };
}

interface AlertData {
  alerts: Alert[];
  empty_message: string;
}

export interface AlertsTab {
  [key: string]: AlertData | string;
  upcoming: AlertData;
  current: AlertData;
  all: AlertData;
  initial_selected: string;
}

export interface Tab {
  badge?: TabBadge;
  class: string;
  href: string;
  id: string;
  name: string;
  selected: string;
}

export interface TabBadge {
  content: string;
  aria_label: string;
  class?: string;
}

export interface RetailLocation {
  address: string;
  latitude: number;
  longitude: number;
  name: string;
  payment: string[];
  phone: string;
}

export interface RetailLocationWithDistance {
  distance: string;
  location: RetailLocation;
}

export interface RouteWithDirections {
  route: EnhancedRoute;
  directions: Direction[];
}
