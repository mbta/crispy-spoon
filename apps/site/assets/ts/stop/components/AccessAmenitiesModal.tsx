import React from "react";
import { Alert, Facility } from "../../__v3api";
import Alerts from "../../components/Alerts";
import { hasFacilityAlert } from "../../models/alert";
import { AmenityModal, AmenityLink } from "./amenities/AmenityCard";

const AccessAmenitiesModal = ({
  stopName,
  alerts,
  facilities,
  facilityType
}: {
  stopName: string;
  alerts: Alert[];
  facilities: Facility[];
  facilityType: "Elevators" | "Escalators";
}): JSX.Element => {
  const hasFacilities = facilities ? facilities.length > 0 : false;
  const linkText =
    facilityType === "Elevators"
      ? "Report an elevator issue"
      : "Report an escalator issue";
  return (
    <>
      {hasFacilities && (
        <AmenityModal headerText={`${facilityType} at ${stopName}`}>
          <Alerts alerts={alerts} />
          <h2>Elevator Status</h2>

          <table className="access-amenities-table">
            <tbody>
              <tr className="access-amenities-header access-amenities-row">
                <th className="access-amenities-name">Elevator</th>
                <th className="status">Status</th>
              </tr>
              {facilities?.map(facility => {
                return (
                  <tr
                    key={facility.id}
                    className="access-amenities-row access-amenities-text"
                  >
                    <td className="access-amenities-name">
                      {facility.attributes.short_name}
                    </td>
                    {hasFacilityAlert(facility.id, alerts) ? (
                      <td className="status">
                        <i className="fa-solid fa-circle amenity-status amenity-out" />
                        Out of Order
                      </td>
                    ) : (
                      <td className="status">
                        <i className="fa-solid fa-circle amenity-status amenity-working" />
                        Working
                      </td>
                    )}
                  </tr>
                );
              })}
            </tbody>
          </table>
          <AmenityLink
            url="/accessibility/trip-planning"
            text="Plan an accessible trip"
          />
          <AmenityLink url="/customer-support" text={linkText} />
        </AmenityModal>
      )}
    </>
  );
};
export default AccessAmenitiesModal;
