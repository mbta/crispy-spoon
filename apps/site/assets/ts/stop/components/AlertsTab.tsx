import React, { useState, ReactElement } from "react";
import { AlertsTab as AlertsTabType, AlertData } from "./__stop";
import Alerts from "../../components/Alerts";
import { handleReactEnterKeyPress } from "../../helpers/keyboard-events";

interface Props {
  alertsTab: AlertsTabType;
}

const selectedClass = (filter: string, selected: string): string =>
  selected === filter ? "m-alerts__time-filter--selected" : "";

const timeFilter = (
  name: string,
  title: string,
  currentFilter: string,
  onClick: React.MouseEventHandler
): ReactElement<HTMLElement> => (
  <div
    role="button"
    id={name}
    tabIndex={0}
    className={`m-alerts__time-filter ${selectedClass(name, currentFilter)}`}
    onClick={onClick}
    onKeyPress={e => handleReactEnterKeyPress(e, onClick)}
  >
    {title}
  </div>
);

const AlertsTab = ({ alertsTab }: Props): ReactElement<HTMLElement> => {
  // eslint-disable-next-line camelcase
  const initialFilter: string = alertsTab.initial_selected || "all";
  const [filter, changeFilter] = useState(initialFilter);

  return (
    <div className="container">
      <div className="page-section station">
        <div className="page-section">
          <div className="row">
            <div className="col-xs-12 col-lg-3">
              <h3 className="h3">Filter by type</h3>
              <div className="m-alerts__time-filters">
                {timeFilter("all", "All Alerts", filter, () =>
                  changeFilter("all")
                )}
                {timeFilter("current", "Current Alerts", filter, () =>
                  changeFilter("current")
                )}
                {timeFilter("upcoming", "Planned Service Alerts", filter, () =>
                  changeFilter("upcoming")
                )}
              </div>
            </div>
            <div className="col-xs-12 col-lg-8 col-lg-offset-1">
              <h2 className="h2">Alerts</h2>
              <div>
                {/* eslint-disable */}
                {(alertsTab[filter] as AlertData).alerts.length ? (
                  <Alerts alerts={(alertsTab[filter] as AlertData).alerts} />
                ) : (
                  <div className="callout">
                    {(alertsTab[filter] as AlertData).empty_message}
                  </div>
                )}
                {/* eslint-enable */}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default AlertsTab;
