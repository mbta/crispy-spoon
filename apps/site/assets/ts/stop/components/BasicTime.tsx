import React from "react";
import parse from "date-fns/parse";
import { formatRelativeTime, formatToBostonTime } from "../../helpers/date";

export interface BasicTimeProps {
  displayType: "relative" | "absolute";
  strikethrough?: boolean;
  time: Date | undefined;
  targetDate?: Date;
}
const BasicTime = ({
  displayType,
  strikethrough,
  time,
  targetDate
}: BasicTimeProps): JSX.Element | null => {
  if (!time) return null;

  const formattedTime =
    displayType === "relative"
      ? formatRelativeTime(time, targetDate)
      : formatToBostonTime(time, "h:mm aa");

  // displayType === "relative" might produce an absolute-formatted time if it's
  // more than an hour away. parse the result to find the truth.
  const parsedTime = parse(formattedTime, "h:mm aa", new Date()).getTime();
  const isAbsolute = Number.isNaN(parsedTime) ? "relative" : "absolute";
  // add data attribute to be used for later manipulation and filtering
  const dataType: "relative" | "absolute" | "imminent" =
    formattedTime === "<1 minute away" ? "imminent" : isAbsolute;
  return (
    <time
      data-type={dataType}
      dateTime={time.toISOString()}
      className={strikethrough ? "strikethrough" : ""}
    >
      {formattedTime}
    </time>
  );
};

export default BasicTime;
