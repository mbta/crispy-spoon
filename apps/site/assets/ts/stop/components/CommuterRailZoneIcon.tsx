import React, { ReactElement } from "react";
import { clickRoutePillAction, Dispatch } from "../state";

const crZone = (
  zoneNumber?: string,
  dispatch?: Dispatch
): ReactElement<HTMLElement> | false => {
  const content = `Zone ${zoneNumber}`;

  let tag = (
    <div className="m-stop-page__header-feature">
      <span className="m-stop-page__icon c-icon__cr-zone">{content}</span>
    </div>
  );

  if (dispatch) {
    tag = (
      <a
        href="#route-card-list"
        onClick={() =>
          dispatch && dispatch(clickRoutePillAction("commuter_rail"))
        }
        className="m-stop-page__header-feature"
      >
        <span className="m-stop-page__icon c-icon__cr-zone">{content}</span>
      </a>
    );
  }

  return !!zoneNumber && zoneNumber.length > 0 && tag;
};

const CommuterRailZoneIcon = ({
  zoneNumber,
  dispatch
}: {
  zoneNumber?: string;
  dispatch?: Dispatch;
}): ReactElement<HTMLElement> => {
  return <>{crZone(zoneNumber, dispatch)}</>;
};

export { CommuterRailZoneIcon as default };
