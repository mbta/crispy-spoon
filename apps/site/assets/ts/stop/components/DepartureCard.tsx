import { reject } from "lodash";
import React, { ReactElement } from "react";
import { Alert, Route } from "../../__v3api";
import { routeName, routeToModeIcon } from "../../helpers/route-headers";
import { routeBgClass } from "../../helpers/css";
import renderSvg from "../../helpers/render-svg";
import useDepartureRow from "../../hooks/useDepartureRow";
import { DepartureInfo } from "../../models/departureInfo";
import { allAlertsForDirection } from "../../models/alert";
import { departureInfoInRoutePatterns } from "../../helpers/departureInfo";
import { isACommuterRailRoute, isSubwayRoute } from "../../models/route";
import DepartureTimes from "./DepartureTimes";
import {
  RoutePatternGroup,
  RoutePatternWithPolyline,
  sortedGroupedRoutePatterns
} from "../../models/route-patterns";

const isNoncanonicalAndNoDepartures = (
  routePatterns: RoutePatternWithPolyline[],
  departures: DepartureInfo[]
): boolean => {
  const isNonCanonical = !routePatterns.find(rp => !!rp.canonical);
  return isNonCanonical && departures.length === 0;
};

const DepartureCard = ({
  alertsForRoute,
  departuresForRoute,
  routePatternsByHeadsign,
  route
}: {
  alertsForRoute: Alert[];
  departuresForRoute: DepartureInfo[];
  routePatternsByHeadsign: RoutePatternGroup;
  route: Route;
}): ReactElement<HTMLElement> | null => {
  const { setRow } = useDepartureRow([route]);

  let sortedRoutePatternsByHeadsign = sortedGroupedRoutePatterns(
    routePatternsByHeadsign
  );

  if (isSubwayRoute(route)) {
    // remove certain headsigns based on route pattern and departures
    sortedRoutePatternsByHeadsign = reject(
      sortedRoutePatternsByHeadsign,
      entry => {
        const [, { route_patterns: routePatterns }] = entry;
        const departures = departuresForRoute.filter(d =>
          departureInfoInRoutePatterns(d, routePatterns)
        );
        return isNoncanonicalAndNoDepartures(routePatterns, departures);
      }
    );
    // don't render a route card if there's no headsigns left to show
    if (sortedRoutePatternsByHeadsign.length === 0) return null;
  }

  return (
    <li className="departure-card">
      <a
        className={`departure-card__route ${routeBgClass(route)} notranslate`}
        href={`/schedules/${route.id}`}
        data-turbolinks="false"
      >
        {renderSvg("c-svg__icon", routeToModeIcon(route), true)}{" "}
        {routeName(route)}
      </a>
      {sortedRoutePatternsByHeadsign.map(
        ([
          headsign,
          { direction_id: directionId, route_patterns: routePatterns }
        ]) => {
          const onClick = (): void =>
            setRow({
              routeId: route.id,
              directionId: directionId.toString(),
              headsign
            });

          return (
            <DepartureTimes
              key={headsign}
              alertsForDirection={allAlertsForDirection(
                alertsForRoute,
                directionId
              )}
              headsign={headsign}
              departures={departuresForRoute.filter(d =>
                departureInfoInRoutePatterns(d, routePatterns)
              )}
              onClick={onClick}
              isCR={isACommuterRailRoute(route)}
              hasService={routePatterns.length !== 0}
            />
          );
        }
      )}
    </li>
  );
};

export default DepartureCard;
