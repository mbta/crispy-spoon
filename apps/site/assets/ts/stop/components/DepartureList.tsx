import React, { ReactElement } from "react";
import { filter } from "lodash";
import { Alert, DirectionId, Route, Stop, Trip } from "../../__v3api";
import { DepartureInfo } from "../../models/departureInfo";
import { SUBWAY, departuresListFromInfos } from "../../helpers/departureInfo";
import { routeBgClass } from "../../helpers/css";
import { routeName, routeToModeIcon } from "../../helpers/route-headers";
import renderSvg from "../../helpers/render-svg";
import { isSuppressiveAlert } from "../../models/alert";
import Alerts from "../../components/Alerts";
import { isACommuterRailRoute } from "../../models/route";

interface DepartureListProps {
  route: Route;
  stop: Stop;
  departures: DepartureInfo[];
  directionId: DirectionId;
  headsign: string;
  alerts: Alert[];
  hasService: boolean;
  targetDate?: Date | undefined;
}

const displayNoUpcomingTrips = (
  message = "No more trips today"
): JSX.Element => {
  return (
    <div className="c-alert-item--low m-8 d-flex justify-content-center align-items-center pb-40 pt-40">
      {message}
    </div>
  );
};

const DepartureList = ({
  route,
  stop,
  departures,
  directionId,
  headsign,
  alerts,
  hasService,
  targetDate
}: DepartureListProps): ReactElement<HTMLElement> => {
  const isCR = isACommuterRailRoute(route);

  // don's show cancelled departures for subway
  const modeSpecificDepartures: DepartureInfo[] = filter(
    departures,
    (d: DepartureInfo) => !(d.isCancelled && d.routeMode === SUBWAY)
  );
  const numPredictions = modeSpecificDepartures.filter(d => !!d.prediction)
    .length;
  const alertsShouldSuppressDepartures = alerts.some(alert =>
    isSuppressiveAlert(alert, numPredictions)
  );
  const tripForSelectedRoutePattern: Trip | undefined =
    modeSpecificDepartures[0]?.trip;

  const noTrips =
    modeSpecificDepartures.length === 0 && displayNoUpcomingTrips();
  const noService = !hasService && displayNoUpcomingTrips("No trips today");
  const noServiceOrNoTrips = noService || noTrips;

  return (
    <>
      <div className="stop-departures departure-list-header">
        <div className={`departure-card__route ${routeBgClass(route)}`}>
          <div className="notranslate">
            {renderSvg("c-svg__icon", routeToModeIcon(route), true)}{" "}
            {routeName(route)}
          </div>
          <a
            className="open-schedule"
            href={`/schedules/${route.id}/line?schedule_direction[direction_id]=${directionId}&schedule_direction[variant]=${tripForSelectedRoutePattern?.route_pattern_id}&schedule_finder[direction_id]=${directionId}&schedule_finder[origin]=${stop.id}`}
            data-turbolinks="false"
          >
            View all schedules
          </a>
        </div>
      </div>
      <h2 className="departure-list__sub-header">
        <div className="departure-list__origin-stop-name notranslate">
          {stop.name} to
        </div>
        <div className="departure-list__headsign notranslate">{headsign}</div>
      </h2>
      {alerts.length ? <Alerts alerts={alerts} /> : null}
      {noServiceOrNoTrips ||
        (!alertsShouldSuppressDepartures && (
          <ul className="stop-routes__departures list-unstyled">
            {departuresListFromInfos(modeSpecificDepartures, isCR, targetDate)}
          </ul>
        ))}
    </>
  );
};

export default DepartureList;
