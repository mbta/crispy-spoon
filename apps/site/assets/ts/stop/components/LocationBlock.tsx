import React, { ReactElement } from "react";
import RoutePillList from "./RoutePillList";
import { Stop } from "../../__v3api";
import { TypedRoutes } from "./__stop";
import renderSvg from "../../helpers/render-svg";

import streetViewSvg from "../../../static/images/icon-street-view-default.svg";
import ExternalRiptaConnection from "./ExternalRiptaConnection";

interface Props {
  routes: TypedRoutes[];
  stop: Stop;
  streetViewUrl: string | null;
}

const addressOrMunicipality = (stop: Stop): ReactElement | null => {
  if (stop.address)
    return (
      <div className="m-stop-page__location">
        <h3 className="u-small-caps">Address</h3>
        <div className="h3">{stop.address}</div>
      </div>
    );

  if (stop.municipality)
    return (
      <div className="m-stop-page__location">
        <h3 className="u-small-caps">City</h3>
        <div className="h3">{stop.municipality}</div>
      </div>
    );

  return null;
};

const latLngString = (stop: Stop): string =>
  `${stop.latitude},${stop.longitude}`;

const streetViewLink = (stop: Stop): string =>
  `https://www.google.com/maps/@?api=1&map_action=pano&viewpoint=${latLngString(
    stop
  )}`;

const LocationBlock = ({
  routes,
  stop,
  streetViewUrl
}: Props): ReactElement<HTMLElement> => (
  <div className="m-stop-page__location-block">
    {addressOrMunicipality(stop)}
    <div className="m-stop-page__location-links">
      <div className="m-stop-page__location-link">
        <a
          href={`/trip-planner/from/${latLngString(stop)
            .replace(/\s+/g, "-")
            .toLowerCase()},${stop.name.replace(/\//g, "%2F")}`}
          className="btn btn-primary"
          target="_blank"
          rel="noopener noreferrer"
        >
          Plan a trip from this station
        </a>
      </div>
      <div className="m-stop-page__location-link">
        <a
          href={streetViewUrl || streetViewLink(stop)}
          target="_blank"
          rel="noopener noreferrer"
        >
          {renderSvg(
            "c-svg__icon-street-view m-stop-page__street-view-icon",
            streetViewSvg
          )}
          Street View
        </a>
      </div>
    </div>
    <RoutePillList routes={routes} />
    <ExternalRiptaConnection stop={stop} />
  </div>
);

export default LocationBlock;
