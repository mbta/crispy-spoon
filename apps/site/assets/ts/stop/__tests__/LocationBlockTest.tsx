import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import stopData from "./stopData.json";
import { StopPageData } from "../components/__stop";
import { Stop } from "../../__v3api";
import LocationBlock from "../components/LocationBlock";

const data = JSON.parse(JSON.stringify(stopData)) as StopPageData;

it("renders", () => {
  document.body.innerHTML =
    '<div><div id="react-root"><div id="test"></div></div></div>';

  const tree = renderer
    .create(
      <LocationBlock
        routes={data.routes}
        stop={data.stop}
        streetViewUrl={null}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("uses lat/lng when window.encodeURIComponent isn't available", () => {
  expect(data.stop.address).toEqual("700 Atlantic Ave, Boston, MA 02110");
  expect(data.stop.latitude).toEqual(42.352271);
  expect(data.stop.longitude).toEqual(-71.055242);
  const encodedAddress = window.encodeURIComponent(data.stop.address!);
  const latLng = `${data.stop.latitude},${data.stop.longitude}`;
  expect(
    shallow(<LocationBlock routes={[]} stop={data.stop} streetViewUrl={null} />)
      .find(".btn.btn-primary")
      .prop("href")
  ).toEqual(`/trip-planner/to/south-station`);

  expect(
    shallow(<LocationBlock routes={[]} stop={data.stop} streetViewUrl={null} />)
      .find(".btn.btn-primary")
      .prop("href")
  ).toEqual(`/trip-planner/to/south-station`);
});

it("falls back to municipality if stop has no address", () => {
  const stopWithoutAddress = { ...data.stop, address: null };

  const wrapper = shallow(
    <LocationBlock routes={[]} stop={stopWithoutAddress} streetViewUrl={null} />
  );

  expect(wrapper.find(".m-stop-page__location .h3").text()).toEqual("Boston");
});

it("does not render location if stop has no address or municipality", () => {
  const stopWithoutLocation = {
    ...data.stop,
    address: null,
    municipality: null
  };

  const wrapper = shallow(
    <LocationBlock
      routes={[]}
      stop={stopWithoutLocation}
      streetViewUrl={null}
    />
  );

  expect(wrapper.find(".m-stop-page__location").exists()).toEqual(false);
  expect(wrapper.find(".m-stop-page__location-links").exists()).toEqual(true);
});
