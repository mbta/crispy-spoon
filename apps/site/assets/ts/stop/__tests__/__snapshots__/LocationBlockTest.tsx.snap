// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="m-stop-page__location-block"
>
  <div
    className="m-stop-page__location"
  >
    <h3
      className="u-small-caps"
    >
      Address
    </h3>
    <div
      className="h3"
    >
      700 Atlantic Ave, Boston, MA 02110
    </div>
  </div>
  <div
    className="m-stop-page__location-links"
  >
    <div
      className="m-stop-page__location-link"
    >
      <a
        className="btn btn-primary"
        href="/trip-planner/from/42.352271,-71.055242,South Station"
        rel="noopener noreferrer"
        target="_blank"
      >
        Plan a trip from this station
      </a>
    </div>
    <div
      className="m-stop-page__location-link"
    >
      <a
        href="https://www.google.com/maps/@?api=1&map_action=pano&viewpoint=42.352271,-71.055242"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          aria-hidden="true"
          className="notranslate c-svg__icon-street-view m-stop-page__street-view-icon"
          dangerouslySetInnerHTML={
            Object {
              "__html": "SVG",
            }
          }
        />
        Street View
      </a>
    </div>
  </div>
  <div
    className="m-route-pills"
  >
    <div
      className="m-route-pills--subway"
    >
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--red-line
    "
        href="/schedules/Red"
        rel="noreferrer"
        target="_self"
      >
        Red Line
      </a>
    </div>
    <div
      className="m-route-pills--bus"
    >
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--silver-line
    "
        href="/schedules/741"
        rel="noreferrer"
        target="_self"
      >
        SL1
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--silver-line
    "
        href="/schedules/742"
        rel="noreferrer"
        target="_self"
      >
        SL2
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--silver-line
    "
        href="/schedules/743"
        rel="noreferrer"
        target="_self"
      >
        SL3
      </a>
    </div>
    <div
      className="m-route-pills--commuter_rail"
    >
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
        href="/schedules/CR-Fairmount"
        rel="noreferrer"
        target="_self"
      >
        Fairmount Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
        href="/schedules/CR-Worcester"
        rel="noreferrer"
        target="_self"
      >
        Framingham/​Worcester Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
        href="/schedules/CR-Franklin"
        rel="noreferrer"
        target="_self"
      >
        Franklin Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
        href="/schedules/CR-Greenbush"
        rel="noreferrer"
        target="_self"
      >
        Greenbush Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
        href="/schedules/CR-Kingston"
        rel="noreferrer"
        target="_self"
      >
        Kingston/​Plymouth Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
        href="/schedules/CR-Middleborough"
        rel="noreferrer"
        target="_self"
      >
        Middleborough/​Lakeville Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
        href="/schedules/CR-Needham"
        rel="noreferrer"
        target="_self"
      >
        Needham Line
      </a>
      <a
        className="
      m-stop-page__header-feature
      m-stop-page__header-description
      u-small-caps
      
      u-bg--commuter-rail
    "
        href="/schedules/CR-Providence"
        rel="noreferrer"
        target="_self"
      >
        Providence/​Stoughton Line
      </a>
    </div>
  </div>
</div>
`;
